### PR TITLE
Add Prometheus Exporter - Utils Unit Tests

### PR DESCRIPTION
**Pull request by CunjunWang**
This PR implements the unit tests for utils in the exporter/prometheus/test folder for the Prometheus Exporter project. The C++ file prometheus_exporter_utils_test.cc implements all the unit test suites for the helper functions that will be called when we translate a OpenTelemetry (OTLP) metric to a Prometheus metric. 

Note to reviewers: 
1. Please keep in mind that there are other PRs related to this one so codecov tests may not fully pass (yet) until other PRs are merged.
2. The Prometheus Exporter Utils is still under development, so we will add more test cases as more code is implemented.

### CODE REVIEW

File: exporters/prometheus/test/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+foreach (
+        testname
+        prometheus_exporter_utils_test
+        prometheus_collector_test
+        prometheus_exporter_test)
+    add_executable(${testname} "${testname}.cc")
+    target_link_libraries(
+            ${testname} ${GTEST_BOTH_LIBRARIES} ${CMAKE_THREAD_LIBS_INIT}
+            prometheus_exporter prometheus-cpp::pull)
+    gtest_add_tests(TARGET ${testname} TEST_PREFIX exporter. TEST_LIST ${testname})
```
Comment (pyohannes) [inline exporters/prometheus/test/CMakeLists.txt:24]: Could you combine this unit tests with the piece of code that is tested into one PR? And provide Bazel build support for it, as well as hooking it into the global CMake build by adding the prometheus directory as a subdirectory to `exporters/CMakeLists.txt`?

I would greatly help the reviewing process to have those tests running in the CI.

Comment (CunjunWang) [reply exporters/prometheus/test/CMakeLists.txt:24]: Yes, I can do that, but I am worrying that the combined PR will be so big. We have three unit test files that contain more than 1000 lines. Can we have such a large PR?

Comment (pyohannes) [reply exporters/prometheus/test/CMakeLists.txt:24]: I think it's good to have each PR buildable, with unit tests that cover the added new functionality. If the only way to ensure that is a bigger PR, than I think it's better to have a bigger PR.

Maybe you can have three PRs: first the exporter utils with unit tests, second the collector with unit tests based on the first, and third the exporter with unit tests based on the second. We can then review and merge from first to third, rebasing later ones when previous ones got merged.

**Comment by CunjunWang** [comment]
Refactored in #280

**Pull request closed without merging**